x/auction/client/cli/tx.go

```diff
@@ -0,0 +1,180 @@
+
+Example:
+  $ %[1]s tx %[2]s place-bid auctionName 123 1000000 --from mykey
+`, version.AppName, types.ModuleName),
+		),
+		Args: cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			sellingTokenAmount, err := strconv.ParseUint(args[1], 10, 64)
+			if err != nil {
+				return fmt.Errorf("cannot parse sellingTokenAmount as uint64 from '%s': %w", args[1], err)
+			}
+
+			paymentTokenAmount, err := strconv.ParseUint(args[2], 10, 64)
+			if err != nil {
+				return fmt.Errorf("cannot parse paymentTokenAmount as uint64 from '%s': %w", args[2], err)
+			}
+
+			msg := types.NewMsgPlaceBid(
+				clientCtx.GetFromAddress().String(),
+				args[0],
+				sellingTokenAmount,
+				paymentTokenAmount,
+			)
+
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func CmdCreateAuction() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create-auction [name] [selling-denom] [payment-denom] [enabled] [price-multiplier] [min-bid-amount] [beneficiary]",
+		Short: "Create a new auction",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Create a new auction for a specific token.
+
+Example:
+  $ %[1]s tx %[2]s create-auction my-auction ibc/DEADBEEF true 0.95 1000000 --from admin
+`, version.AppName, types.ModuleName),
+		),
+		Args: cobra.ExactArgs(4),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			enabled := args[3] == "true"
+
+			minBidAmount, err := strconv.ParseUint(args[3], 10, 64)
+			if err != nil {
+				return fmt.Errorf("cannot parse minBidAmount as uint64 from '%s': %w", args[3], err)
+			}
+
+			msg := types.NewMsgCreateAuction(
+				clientCtx.GetFromAddress().String(),
+				args[0],
+				types.AuctionType_AUCTION_TYPE_FCFS,
+				args[1],
+				args[2],
+				enabled,
+				args[4],
+				minBidAmount,
+				args[6],
+			)
+
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func CmdUpdateAuction() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update-auction [name] [enabled] [price-multiplier] [min-bid-amount] [beneficiary]",
+		Short: "Update an existing auction",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Update an existing auction's parameters.
+
+Example:
+  $ %[1]s tx %[2]s update-auction auctionName true 0.97 500000 --from admin
+`, version.AppName, types.ModuleName),
+		),
+		Args: cobra.ExactArgs(4),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			enabled := args[1] == "true"
```

Comment:
@sampocs sampocs Dec 17, 2024

nit: i think you can do ParseBool

assafmo Dec 17, 2024

parseBool is more lines and less easter eggs
e.g. "stfu" => false can sometimes brighten your day
otoh "tru" => false can f up your day

Reply:
@sampocs4419b8ca808217c1776954e9f36c8f14914b56d6